plugins/module_utils/openshift_builds.py

```diff
@@ -0,0 +1,662 @@
+
+        changed = False
+        result = []
+        for build in builds_to_cancel:
+            # Set cancelled to true
+            build["status"]["cancelled"] = True
+            name = build["metadata"]["name"]
+            changed = True
+            cancelled_build = build
+            if not self.check_mode:
+                try:
+                    content_type = "application/json"
+                    cancelled_build = self.client.request(
+                        "PUT",
+                        "/apis/build.openshift.io/v1/namespaces/{0}/builds/{1}".format(
+                            namespace, name
+                        ),
+                        body=build,
+                        content_type=content_type,
+                    ).to_dict()
+                except DynamicApiError as exc:
+                    self.fail_json(
+                        msg="Failed to cancel Build %s/%s due to: %s" % (namespace, name, exc),
+                        reason=exc.reason,
+                        status=exc.status
+                    )
+                except Exception as e:
+                    self.fail_json(
+                        msg="Failed to cancel Build %s/%s due to: %s" % (namespace, name, e)
+                    )
+            result.append(cancelled_build)
+
+        # Make sure the build phase is really cancelled.
+        def _wait_until_cancelled(build):
+            timeout = 30
+            if build["spec"].get("strategy", {}).get("jenkinsPipelineStrategy"):
+                # Bump the timeout in case we have to wait for Jenkins
+                # to come up so that the sync plugin can actually change the phase
+                timeout = timeout + 3 * 60
+            start = datetime.now()
+            last_phase = None
+            name = build["metadata"]["name"]
+            while (datetime.now() - start).seconds < timeout:
+                params = dict(
+                    kind=kind,
+                    api_version=api_version,
+                    name=name,
+                    namespace=namespace
+                )
+                resource = self.kubernetes_facts(**params).get("resources", [])
+                if len(resource) == 0:
+                    return None, "Build %s/%s not found" % (namespace, name)
+                resource = resource[0]
+                last_phase = resource["status"]["phase"]
+                if last_phase == "Cancelled":
+                    return resource, None
+            return None, "Build %s/%s is still %s" % (namespace, name, last_phase)
+
+        if not self.check_mode:
+            wait_result = []
+            for build in result:
+                ret, err = _wait_until_cancelled(build)
+                if err:
+                    self.exit_json(msg=err)
+                wait_result.append(ret)
+            result = wait_result
+
+        if self.params.get("restart"):
+            self.start_build()
+            # started = []
```

Comment:
Why do we want to keep this commented block?

Reply:
useless, will remove